Automattic/wp-calypso
# Stats: Add tier quantity information to checkout

Related to #85326 (based on #84837)

## Proposed Changes

* Add a condition to `getLabel` to display additional information for Jetpack Stats tiered plans, making it clear to the user which plan they are purchasing.

## Testing Instructions



* Checkout the branch and make sure you sandbox the store and have `D129634` diff applied
* Go to `/checkout/<blog_slug>/jetpack_stats_yearly` and confirm nothing changes
* Visit `/checkout/<blog_slug>/jetpack_stats_yearly:-q-<your quantity>` and confirm the number of requests is part of the product title
* Purchase the plan using a test card or credits
* Go to `/me/purchases`, and click to see the new purchase
* Confirm that you see the purchased quantity in the title

After the update:
<img width="672" alt="SCR-20231219-kuqt" src="https://github.com/Automattic/wp-calypso/assets/112354940/d9631567-9cb3-4a2f-ab4e-8b2d27970715">

<img width="916" alt="SCR-20231219-kvdw" src="https://github.com/Automattic/wp-calypso/assets/112354940/669c3ab1-b23e-44ac-a621-7b7067b13090">


## Pre-merge Checklist
